src/common/models/Projection.ts

```diff
@@ -403,10 +413,13 @@ export class Projection {
       currDate = currDate.clone().add(1, 'days');
     }
 
+    // only keep futureDaysToInclude days ahead of today
+    const now = new Date();
+    const days = dates.findIndex(date => date > now) + futureDaysToInclude;
     return {
-      timeseries,
-      actualsTimeseries,
-      dates,
+      timeseries: timeseries.slice(0, days),
+      actualTimeseries: actualsTimeseries.slice(0, days),
+      dates: dates.slice(0, days),
```

Comment:
FWIW- Instead of doing this truncation logic here at the end I think you could just change latestDate at the top to:

```
const latestDate = moment().add(futureDaysToInclude, 'days');
```